packages/ceramic-core/src/doctypes/accountLinkHandler.ts

```diff
@@ -0,0 +1,85 @@
+import DoctypeHandler from './doctypeHandler'
+import { DocState, SignatureStatus } from '../document'
+import jsonpatch from 'fast-json-patch'
+import { validateLink } from '3id-blockchain-utils'
+
+const DOCTYPE = 'account-link'
+
+function head (log: Array<string>): string {
+  return log[log.length - 1]
+}
+
+class AccountLinkHandler extends DoctypeHandler {
+  constructor () {
+    super(DOCTYPE)
+  }
+
+  async applyRecord (record: any, cid: string, state?: DocState): Promise<DocState> {
+    let newState
+    if (record.doctype) {
+      newState = await this.applyGenesis(record)
+    } else if (record.proof) {
+      newState = await this.applyAnchor(record, state)
+    } else if (record.content || record.owners) {
+      newState = await this.applySigned(record, state)
+    }
+    newState.log.push(cid)
+    return newState
+  }
+
+  async makeRecord (state: DocState, newContent: any): Promise<any> {
```

Comment:
So newContent here is the proof right? I would just put this proof directly as the content of this record.